src/Vision/V1/ImageAnnotatorClient.php

```diff
@@ -30,13 +30,458 @@
 
 namespace Google\Cloud\Vision\V1;
 
+use Google\ApiCore\ApiException;
+use Google\ApiCore\ArrayTrait;
+use Google\ApiCore\RetrySettings;
+use Google\Cloud\Vision\VisionHelpersTrait;
 use Google\Cloud\Vision\V1\Gapic\ImageAnnotatorGapicClient;
+use InvalidArgumentException;
 
 /**
  * {@inheritdoc}
  */
 class ImageAnnotatorClient extends ImageAnnotatorGapicClient
 {
-    // This class is intentionally empty, and is intended to hold manual
-    // additions to the generated {@see ImageAnnotatorClientImpl} class.
+    use VisionHelpersTrait;
+
+    /**
+     * Creates an Image object that can be used as part of an image annotation request.
+     *
+     * Example:
+     * ```
+     * //[snippet=resource]
+     * $imageResource = fopen('path/to/image.jpg', 'r');
+     * $image = $imageAnnotatorClient->createImageObject($imageResource);
+     * $response = $imageAnnotatorClient->faceDetection($image);
+     * ```
+     *
+     * ```
+     * //[snippet=data]
+     * $imageData = file_get_contents('path/to/image.jpg');
+     * $image = $imageAnnotatorClient->createImageObject($imageData);
+     * $response = $imageAnnotatorClient->faceDetection($image);
+     * ```
+     *
+     * ```
+     * //[snippet=url]
+     * $imageUri = "gs://my-bucket/image.jpg";
+     * $image = $imageAnnotatorClient->createImageObject($imageUri);
+     * $response = $imageAnnotatorClient->faceDetection($image);
+     * ```
+     *
+     * @param  resource|string $imageInput An image to configure with
+     *         the given settings. This parameter will accept a resource, a
+     *         string of bytes, or the URI of an image in a publicly-accessible
+     *         web location.
+     * @return Image
+     * @throws InvalidArgumentException
+     */
+    public function createImageObject($imageInput)
+    {
+        return $this->createImageHelper(Image::class, ImageSource::class, $imageInput);
+    }
+
+    /**
+     * Run image detection and annotation for an image.
+     *
+     * Example:
+     * ```
+     * use Google\Cloud\Vision\V1\Feature;
+     * use Google\Cloud\Vision\V1\Feature_Type;
+     *
+     * $imageResource = fopen('path/to/image.jpg', 'r');
+     * $image = $imageAnnotatorClient->createImageObject($imageResource);
```

Comment:
I am wondering if we could short circuit the factory method and have the first argument of `annotateImage` accept `resource|string`. Are there many cases where users may want to hold on to the `Image` instance after a request for further processing?

Reply:
I think that the factory method has value outside of the annotateImage method - anyone using the BatchAnnotateImages method can also use it.

Reply:
I didn't mean to suggest removing the factory method, just not always requiring its usage.

Reply:
Ah ok. Yep I think that is a good idea - done!